pkgs/development/python-modules/aws-adfs/default.nix

```
@@ -2,12 +2,12 @@
 , pytest, pytestrunner, pytestcov, mock, glibcLocales, lxml, boto3, requests, click, configparser }:
 
 buildPythonPackage rec {
-  version = "1.12.3";
   pname = "aws-adfs";
+  version = "1.17.0";
 
```

Although it builds, I get the following error with python2:
```
[nix-shell:/home/jon/.cache/nix-review/pr-68183]$ ./results/python27Packages.aws-adfs/bin/aws-adfs
Traceback (most recent call last):
  File "/nix/store/0fd75gk77ccl4rv5aniymnw142f9ij2n-python2.7-aws-adfs-1.17.0/bin/.aws-adfs-wrapped", line 7, in <module>
    from aws_adfs.commands import cli
  File "/nix/store/plcsb62sjrljf6ip0ifj8n027wns01ac-python3.7-aws-adfs-1.17.0/lib/python3.7/site-packages/aws_adfs/commands.py", line 9, in <module>
    from . import login
  File "/nix/store/plcsb62sjrljf6ip0ifj8n027wns01ac-python3.7-aws-adfs-1.17.0/lib/python3.7/site-packages/aws_adfs/login.py", line 11, in <module>
    from . import authenticator
  File "/nix/store/plcsb62sjrljf6ip0ifj8n027wns01ac-python3.7-aws-adfs-1.17.0/lib/python3.7/site-packages/aws_adfs/authenticator.py", line 2, in <module>
    import lxml.etree as ET
ImportError: No module named etree
```
```suggestion
  disabled = isPy27;
  
```

---

That's weird - it appears to be python 2 and 3 libraries. Same happens for me on macos 10.13.

Non-nixos linux x86_64, both work fine for me though.

---

actually, i know what this is, my nix-review nix-shell has a python3 env loaded, so the pythonpath is all messed up if i did `nix-shell -f ./nixpkgs -A python2Packages.aw-adfs` then it would probably work fine.

---

Hmm. But as a standalone binary, the environment its run from shouldn't be relevant. Seems like the binary needs to be wrapped.

---

It's a wrapped python script, so the env will populate PYTHONPATH, and python will gladly put that on sys.path at runtime

---

Hmm well I guess this is the difference between `buildPythonApplication` and `buildPythonPackage`.

---

I think i ran into an issue where even a `toPythonApplication` cli tool had this issue

---

This does work on both python2 and python3, but since it's a cmdline devtool I think it should be in all-packages, rather than in python-modules.  I've updated the PR to move it accordingly.

---

i would rather have it in python-modules, and just make an entry in top-level/all-packages that uses the `toPythonApplication` function to wrap it

---

You can and should put it in both by simply using `toPythonApplication` as mentioned above to make the "app" version for `all-packages.nix`.